src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs

```diff
@@ -280,6 +298,17 @@ public string DomainHint
         }
 
         /// <summary>
+        /// Gets or sets whether parameters for the library and version are sent on the query string for this <see cref="OpenIdConnectMessage"/> instance. 
+        /// This value is set to the value of EnableTelemetryParametersByDefault at message creation time.
+        /// </summary>
+        public bool EnableTelemetryParameters { get; set; } = EnableTelemetryParametersByDefault;
+
+        /// <summary>
+        /// Gets or sets whether parameters for the library and version are sent on the query string for all instances of <see cref="OpenIdConnectMessage"/>. 
+        /// </summary>
+        public static bool EnableTelemetryParametersByDefault = true;
```

Comment:
Property implementations can be changed later without breaking callers. Fields can be changed to properties later but it requires recompiling the caller.